dev/com.ibm.ws.security.jwtsso_fat/fat/src/com/ibm/ws/security/jwtsso/fat/ConfigAttributeTests.java

```diff
@@ -122,33 +120,37 @@ public void test_invalidBuilderRef_ltpaFallbackFalse() throws Exception {
-     * Authcache must be disabled in server.xml for this test.
+     * Test the jwtBuilderRef attribute. Specify an existing and valid jwtBuilderRef.
+     * Authentication will fail because the issuer mismatches the consumer, but we should
+     * see evidence in the logs that the customized issuer was presented.
+     * That's all we care about.
      */
+    @AllowedFFDC({ "com.ibm.websphere.security.jwt.InvalidClaimException",
+                   "com.ibm.websphere.security.jwt.InvalidTokenException",
+                   "com.ibm.ws.security.jwt.internal.JwtTokenException",
+                   "com.ibm.websphere.security.WSSecurityException",
+                   "com.ibm.ws.security.authentication.AuthenticationException" })
     @Mode(TestMode.LITE)
     @Test
     public void test_validBuilderRef() throws Exception {
         server.reconfigureServer(JwtFatConstants.COMMON_CONFIG_DIR + "/server_testgoodbuilder.xml");
 
         String issuer = "https://flintstone:19443/jwt/defaultJWT";
 
+        Page response = actions.invokeUrl(testName.getMethodName(), protectedUrl); // get back the login page
+
+        // now confirm we got the login page
         Expectations expectations = new Expectations();
         expectations.addExpectations(CommonExpectations.successfullyReachedLoginPage(TestActions.ACTION_INVOKE_PROTECTED_RESOURCE));
-
-        Page response = actions.invokeUrl(testName.getMethodName(), webClient, protectedUrl); // get back the login page
         validationUtils.validateResult(response, TestActions.ACTION_INVOKE_PROTECTED_RESOURCE, expectations);
 
-        expectations.addExpectations(CommonExpectations.successfullyReachedUrl(TestActions.ACTION_SUBMIT_LOGIN_CREDENTIALS, protectedUrl));
-        expectations.addExpectations(CommonExpectations.jwtCookieExists(TestActions.ACTION_SUBMIT_LOGIN_CREDENTIALS, webClient, JwtFatConstants.JWT_COOKIE_NAME));
-        expectations.addExpectations(CommonExpectations.getResponseTextExpectationsForJwtCookie(TestActions.ACTION_SUBMIT_LOGIN_CREDENTIALS, JwtFatConstants.JWT_COOKIE_NAME,
-                                                                                                defaultUser, Pattern.quote(issuer)));
-        expectations.addExpectations(CommonExpectations.getJwtPrincipalExpectations(TestActions.ACTION_SUBMIT_LOGIN_CREDENTIALS, defaultUser, Pattern.quote(issuer)));
-        expectations.addExpectations(CommonExpectations.responseTextMissingCookie(TestActions.ACTION_SUBMIT_LOGIN_CREDENTIALS, JwtFatConstants.LTPA_COOKIE_NAME));
+        // log in, which should drive building a token.
+        response = actions.doFormLogin(response, defaultUser, defaultPassword);
+
+        String errorMsg = server.waitForStringInLogUsingMark(".*CWWKS6022E: The issuer \\[https://flintstone:19443/jwt/defaultJWT\\]", 100);
+        // CWWKS6022E - issuer not trusted.
+        assertTrue("Did not find expected error message CWWKS6022E in the log", errorMsg != null);
```

Comment:
This surprises me - whenever an error message is emitted without calling `server.addIgnoredErrors(...)` to add it as an expected error message, a test failure should occur for an unexpected error/warning message found in the logs. I'm not sure how you didn't get a build failure.